ts_src/lib/converter/shared/tapBip32Derivation.ts

```diff
@@ -46,16 +46,17 @@ export function makeConverter(
     'masterFingerprint: Buffer; ' +
     'pubkey: Buffer; ' +
     'path: string; ' +
-    'leafHashes: Buffer[]; ' +
+    'leafHashes?: Buffer[]; ' +
```

Comment:
`TapBip32Derivation` requires leafHashes.

Look at the type in the interfaces.ts

Reply:
I think we should also set leafHashes in the interface to optional, because it does not exist in the taproot address of the key-path

Reply:
Or if leafHashes is required, you need to modify the check method to allow empty arrays

Reply:
> if leafHashes is required, you need to modify the check method to allow empty arrays

It does allow empty arrays. `every()` returns true if the array is empty.

Reply:
> > if leafHashes is required, you need to modify the check method to allow empty arrays
> 
> It does allow empty arrays. `every()` returns true if the array is empty.

Sorry, maybe I passed undefined before and triggered the check failed. I have rolled back and just updated the readme